crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -1564,168 +1554,144 @@ where
-                        // argument and without CL.
-                        warn!(target: "consensus::engine", "No fork choice state available");
-                        return None
-                    }
-                };
+        // get the block number of the finalized block, if we have it
+        let newest_finalized = self
+            .blockchain
+            .buffered_header_by_hash(sync_target_state.finalized_block_hash)
+            .map(|header| header.number);
 
-                // Next, we check if we need to schedule another pipeline run or transition
-                // to live sync via tree.
-                // This can arise if we buffer the forkchoice head, and if the head is an
-                // ancestor of an invalid block.
-                //
-                //  * The forkchoice head could be buffered if it were first sent as a `newPayload`
-                //    request.
-                //
-                // In this case, we won't have the head hash in the database, so we would
-                // set the pipeline sync target to a known-invalid head.
-                //
-                // This is why we check the invalid header cache here.
-                let lowest_buffered_ancestor =
-                    self.lowest_buffered_ancestor_or(sync_target_state.head_block_hash);
-
-                // this inserts the head if the lowest buffered ancestor is invalid
-                if self
-                    .check_invalid_ancestor_with_head(
-                        lowest_buffered_ancestor,
-                        sync_target_state.head_block_hash,
-                    )
-                    .is_none()
-                {
-                    // get the block number of the finalized block, if we have it
-                    let newest_finalized = self
-                        .forkchoice_state_tracker
-                        .sync_target_state()
-                        .map(|s| s.finalized_block_hash)
-                        .and_then(|h| self.blockchain.buffered_header_by_hash(h))
-                        .map(|header| header.number);
-
-                    // The block number that the pipeline finished at - if the progress or newest
-                    // finalized is None then we can't check the distance anyways.
-                    //
-                    // If both are Some, we perform another distance check and return the desired
-                    // pipeline target
-                    let pipeline_target = if let (Some(progress), Some(finalized_number)) =
-                        (ctrl.block_number(), newest_finalized)
-                    {
-                        // Determines whether or not we should run the pipeline again, in case the
-                        // new gap is large enough to warrant running the pipeline.
-                        self.can_pipeline_sync_to_finalized(progress, finalized_number, None)
-                    } else {
-                        None
-                    };
```

Comment:
`if let (Some(), Some()) { /* invoke fn that returns an option */ } else { None }` is simply refactored to `.zip().and_then(...)`